src/app/ui-components/selectFeeRate/feeSelectPopup.tsx

```diff
@@ -0,0 +1,304 @@
+    }
+
+    return { knownRates: rates, selected: selectedRate };
+  }, [feeRates, currentFeeRate]);
+
+  const renderFeeItem = (priority: FeePriority, rate: number) => (
+    <FeeItem
+      priority={priority}
+      feeRate={rate}
+      onClick={handleClick(rate.toString())}
+      selected={priority === selected}
+      feeUnits={feeUnits}
+      feeRateUnits={feeRateUnits}
+      fiatUnit={fiatUnit}
+      baseToFiat={baseToFiat}
+      getFeeForFeeRate={getFeeForFeeRate}
+    />
+  );
+
+  const renderFeeSelectors = () => {
+    if (useCustom) {
+      const onCustomValueChange = (e: ChangeEvent<HTMLInputElement>) => {
+        if (inputValidator.test(e.target.value)) setCustomValue(e.target.value);
+      };
+
+      const onApply = () => {
+        if (!hasSufficientFunds) return;
+
+        setFeeRate(`${+customValue}`);
+        onClose();
+      };
+
+      const rateTooLow = !!feeRateLimits?.min && Number(customValue) < feeRateLimits.min;
+      const rateTooHigh = !!feeRateLimits?.max && Number(customValue) > feeRateLimits.max;
+
+      const errorText =
+        // eslint-disable-next-line eqeqeq
+        +customValue == 0
+          ? undefined
+          : !hasSufficientFunds
+          ? t('SEND.INSUFFICIENT_FUNDS')
+          : rateTooLow
+          ? t('SEND.RATE_TOO_LOW', { minFee: feeRateLimits?.min })
+          : rateTooHigh
+          ? t('SEND.RATE_TOO_HIGH', { maxFee: feeRateLimits?.max })
+          : undefined;
+
+      return (
+        <>
+          <Input
+            onChange={onCustomValueChange}
+            value={customValue}
+            placeholder="0"
+            complications={feeRateUnits}
+            hideClear
+          />
+          <SummaryContainer>
+            {customValue && hasSufficientFunds && !isCalculatingTotalFee && (
+              <>
+                <TotalFeeContainer>
+                  <StyledP typography="body_medium_m" color="white_200">
+                    {t('TRANSACTION_SETTING.TOTAL_FEE')}:
+                  </StyledP>
+                  <StyledP typography="body_medium_m" color="white_0">
+                    {customValueTotalFee} {feeUnits}
+                  </StyledP>
+                </TotalFeeContainer>
+                <NumericFormat
+                  value={customValueTotalFeeFiat}
+                  displayType="text"
+                  thousandSeparator
+                  prefix={`~ ${currencySymbolMap[fiatUnit]}`}
+                  suffix={` ${fiatUnit}`}
+                  renderText={(value: string) => (
+                    <StyledP typography="body_medium_m" color="white_200">
+                      {value}
+                    </StyledP>
+                  )}
+                />
+              </>
+            )}
+            {isCalculatingTotalFee && <Spinner />}
+          </SummaryContainer>
+          {errorText && (
+            <StyledP typography="body_medium_m" color="danger_light">
+              {errorText}
+            </StyledP>
+          )}
```

Comment:
this should probably be supplied in feedbacks prop to `<Input>`?
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/79c4aef7-0da7-49cc-a71b-30097e4ac312)